### ISSUE TITLE
[2020 NeurIPS] Early-Learning Regularization Prevents Memorization of Noisy Labels

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#92]
딥러닝 모델이 noisy dataset에서 학습할 때, early learning phase에서는 clean data에 fit하지만 epoch이 증가하면서 noisy data에 대해 memorization이 발생하게 되는 것은 널리 알려졌음

해당 논문은 이에 더 나아가 high-dimensional noisy dataset이라면, simple linear model에서도 위와 똑같은 현상이 나타남을 실험을 통해 보임

또한 simple linear model에서의 gradient에 대한 intuition을 이용하여 `early-learning regularization (ELR)`을 제안하며, 빠른 학습 속도와 매우 좋은 성능을 내면서 hyperparameter에 robust함을 실험을 통해 보임

즉 논문의 2가지 main contribution은 다음과 같음

1. High-dimensional noisy dataset에서는 simple linear model도 memorization 발생
2. `ELR` regularization을 제안하며, 이는 학습 속도가 빠르면서 성능이 좋고 hyperparameter에 robust함

논문의 핵심만 간단히 요약

## 1. Early learning as a general phenomenon of high-dimensional classification

<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195348399-a8375020-3565-418d-8b79-718d9dfb528b.png">
<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195348714-8ac9cfe3-c2e6-462e-983a-2e9ca8eba6d9.png">

위와 같이 x, y를 sample한 후, data를 noisy하게 만듬

이렇게 생성된 noisy dataset을 이용하여 simple linear model을 cross entropy loss로 학습

<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195348005-7ac88ccf-091d-454b-9e5f-a86714509c71.png">

학습한 결과, 기존 딥러닝 모델들과 같이 early-learning phase에서는 early learning phase에서는 clean data에 대해 학습하다가, 학습이 진행되며 noisy data에 대해 memorization이 발생함을 알 수 있음

### 2.1. Gradient analysis of softmax classification from noisy labels

앞에서 관측한 현상이 일어나는 이유에 대해 저자들은 gradient를 통해 분석

> - Early in training, the gradients corresponding to the correctly labeled examples dominate the dynamics, which leads to early progress towards the true optimum
> - But that the gradients corresponding to wrong labels soon become dominant, the classifier simply learns to fit the noisy labels

즉 학습 초기에는 clean data의 gradient가 dominant해서 clean data에 맞게 학습되지만, 어느정도 학습이 된 이후에는 noisy data의 gradient가 dominant해지면서 memorization이 발생하는 것

이를 수식으로 보면 다음과 같음

> <img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195375871-63afee0f-e04b-477a-9ef1-dd86c0331c24.png">
> <img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195472228-f906df11-5b2d-4f46-b7fe-3d716dcc350d.png">
>
> - 학습 초기에는 모든 data에 대한 모델의 prediction p와 y의 차이가 커, 비중이 많은 clean data의 gradient가 dominant하여 올바르게 학습
> - 학습이 진행됨에 따라 clean data에 대한 모델의 prediction p가 y와 가까워지고 noisy data에 대한 모델의 prediction p와 y의 차이가 커져 noisy data의 gradient가 dominant하여 memorization

### 2.2. Early-learning regularization

위를 통해 2가지 조건을 만족하면 noisy data를 memorize하지 않을 것이라 예상할 수 있음

1. Boosting the gradient of examples with clean labels
2. Neutralizing the gradient of examples with false labels

저자들은 semi-supervised learning에서 intuition을 얻어, 각 example마다 target probability vector t가 있으며 t가 optimization process 초기에 noisy label에 overfit하지 않을 것이라는 가정 하에 early-learning regularization (ELR)을 제안함

ELR은 cross entropy loss에 model output p와 target t의 inner product를 maximize하는 regularization term을 붙인 loss이며 자세한 식은 다음과 같음

<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195460079-44ea0b86-4570-454c-adaa-ccd3da57e1f8.png">

ELR loss의 gradient를 구하고 다음과 같이 분석

> <img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195460109-476f328b-479c-4008-bc65-d741ab7a8a9d.png">
> <img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195472097-47b4cc45-f6a2-42c0-9223-f784519f981c.png">
>
> - 학습 초기의 경우, example의 true class c*에 대한 t 값이 커지게 됨
> - t가 클 때 g값을 계산해보면 true class c*에 대해 음수 값을 가지게 됨
> - 즉 학습이 어느정도 진행되었을 때, clean data의 gradient의 p - y term이 0에 가까워져 영향력이 없어지는 것이 아닌 p - y - lambda * abs(g)가 되어 영향력을 계속 유지하게 됨

### 2.3. Target estimation

단순하게 model output을 target으로 사용하지 않고 어떻게 target을 구할 것인지에 따라 2가지 방식으로 나뉨

- ELR : temporal ensembling
- ELR+ : temporal ensembling + weight-averaging + 2-network + mixup data augmentation

<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195471384-a542710b-10f7-41a3-9cd1-762be301aea0.png">
<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195471400-d0d20d92-e871-4f6c-94f2-8b82f61657f8.png">

ELR+에 들어가는 테크닉들이 성능 향상에 유의미한 것을 ablation study를 통해 보임

<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195472038-b9be5eb2-8f35-4ea7-8eda-d4314905594c.png">

## 3. Results
<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195471938-2a8b1eee-4a69-405d-943f-6e74a25292f2.png">
<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195471954-fe8b9dd2-af4f-411b-a796-3bd2d5322e4f.png">
<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195472003-7a544b6d-24ce-4125-81c9-ead2ce5db7d0.png">

많은 비전 dataset에서 성능이 sota거나 2등

<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195472410-a079f003-b712-44dd-86f2-140e23ea23a9.png">

그런데 학습 속도는 빠름

<img width="700" alt="image" src="https://user-images.githubusercontent.com/49643709/195472444-08703673-2f50-4278-8fa6-44eedb125663.png">

Hyperparameter에 그렇게 크게 sensitive하지도 않음

즉 학습 속도가 빠르면서 성능도 잘 나오고 hyperparameter에도 robust한 방식이라는 것
